projects/libs/core-library-typescript/src/subscription-manager/subscription-manager.ts

```diff
@@ -16,7 +26,58 @@ export class SubscriptionManager {
     this.contractProxy = new ERCBillingContractProxy(sidechainConnector, this.config.ethereumContractAddress);
   }
 
-  async getSubscriptionStatus(subscriptionKey: string) {
-    return await this.contractProxy.getSubscription(subscriptionKey);
+  async isSubscriptionValid(subscriptionKey: string): Promise<boolean> {
+    const subscriptionData =  await this.contractProxy.getSubscriptionData(subscriptionKey);
+
+    const now = new Date();
+
+    const rateData = this.getCurrentSubscriptionRateByProfile(subscriptionData.profile, now);
+
+    if (rateData == undefined) {
+      logger.info(`Could not find rate settings for subscription profile ${subscriptionData.profile}`);
+      return false;
+    }
+
+    const beginningOfThisMonth = new Date(`${now.toISOString().substr(0, 7)}-01`).getTime();
+
+    let minimumRequiredFee: number;
+
+    if (subscriptionData.startTime <= beginningOfThisMonth) {
+      minimumRequiredFee = rateData.rate;
+    } else {
+      const numOfDaysThisMonth = new Date(now.getFullYear(), now.getMonth() + 1, 0).getDate();
+      const endOfThisMonth = new Date(`${now.toISOString().substr(0, 7)}-${numOfDaysThisMonth}`).getTime();
+
+      const subscriptionDurationInMilliseconds = endOfThisMonth - subscriptionData.startTime;
+      const numOfMillisecondsThisMonth = endOfThisMonth - beginningOfThisMonth;
+
+      minimumRequiredFee = rateData.rate * (subscriptionDurationInMilliseconds / numOfMillisecondsThisMonth);
+    }
+
+    if (subscriptionData.tokens < minimumRequiredFee) {
```

Comment:
Should probably be <=

Reply:
It's the other way around, if we have less than X tokens we throw an error.